### PR TITLE
Add specs package

### PR DESCRIPTION
**Pull request by maurelian**
### Description

This PR moves specifications from a private repo to here. Things I think reviewers should focus on: 

1. Identifying factually incorrect information.
2. Identifying information that is going to change in the near future (I will add a note in the specs to reflect it).
If you respond with 'Request changes', please indicate what you would require to approve. :)

### Responses to some comments/questions I anticipate

**1. What's up with these files that have no substantive content?** 

I have included a couple files that are effectively empty (or have their content commented out). Per the note in these files, I would like to keep them as a place holder.

**2. Where is the content to describe how _X_ works?**

Please DO leave these comments so I can prioritize the areas most in need of specification. However, IMO completeness is shouldn't be a requirement to merge this in. 

**3. What going on in `/l2geth`?** 

The content there was written by others in the previous repo. I think it's useful to bring this in for the geth team. I'm open to splitting it out into another location, if there is a clear proposal, otherwise would prefer just to get it in here.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (maurelian) [inline .github/CODEOWNERS:4]: I moved `packages/contracts/` up to make the relationship between the specs and the implementation locations more obvious.

File: packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol
Comment (maurelian) [inline packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:323]: Just a comment I noticed needed improvement while doing this work.

File: packages/specs/l2geth/sequencer/README.md
Comment (maurelian) [inline packages/specs/l2geth/sequencer/README.md:1]: `/seqencer` had it's own top level dir in [the specs repo](https://github.com/ethereum-optimism/specs/tree/main/sequencer). I moved it into `/l2geth` here. I supposed I could have done: 
```
mv l2geth/sequencer/* l2geth
rm -rf sequencer
```

But this felt like a more conservative approach to content written by others.

File: packages/specs/package.json
Comment (maurelian) [inline packages/specs/package.json]: 1. I intend for `format:check` to run in CI, but haven't set that up yet. 

2. I know that `yarn prettier --check ./**/*.md` looks funny. Why not just `prettier --check **/*.md`, honestly I'm not sure why, but that and every other variation of this script I tried failed to run on all md files up and down the tree.

**Comment by maurelian** [comment]
Draft very briefly while I do my own review.

File: packages/specs/package.json
Comment (maurelian) [inline packages/specs/package.json:10]: 1. I realize the glob string seems like overkill, but I couldn't get anything else to hit all the md files up and down the tree. 
2. my intention is for `format:check` to run in CI, but I haven't set that up.

File: packages/specs/protocol/components/accounts.md
Comment (maurelian) [inline packages/specs/protocol/components/accounts.md:3]: I hope to complete this document thoroughly in the near future, but would like to include it as a placeholder. 
If it's a blocker, I can remove it.

File: packages/specs/protocol/components/execution.md
Comment (maurelian) [inline packages/specs/protocol/components/execution.md:67]:
```suggestion
```

File: packages/specs/protocol/data-structures.md
Comment (maurelian) [inline packages/specs/protocol/data-structures.md:1]: Feel free to leave a comment here where you think a variable name should change, but IMO actually updating it should be reserved for a new PR.

**Comment by tynes** [comment]
> The content there was written by others in the previous repo. I think it's useful to bring this in for the geth team. I'm open to splitting it out into another location, if there is a clear proposal, otherwise would prefer just to get it in here.

I'm on board with leaving it in and agree with renaming it to `l2geth`

**Comment by tynes** [comment]
I made an update to `l2geth` here: https://github.com/ethereum-optimism/specs/pull/31

I can open a PR against this or wait until this is merged in, whatever is easier for @maurelian getting this in

**Review by gakonst — CHANGES_REQUESTED** [review]
Please:
1. pull this to the top level directory (not under `packages`), and add it to the global yarn workspace.
2. mark the package as private so that it does not get published by accident

Have the `protocol` changes been approved by @ben-chain? https://github.com/ethereum-optimism/specs/pull/30 is still a draft.

Separately:
It seems like `transaction-indexer` refers to the DTL so should be moved outside of `l2geth`, and the `transaction-ingestor` refers to L2geth's `SyncService` so should be renamed to that?

**Comment by maurelian** [comment]
> Have the protocol changes been approved by @ben-chain? ethereum-optimism/specs#30 is still a draft.

There should not be protocol changes in here yet. For now I've aimed to describe the protocol as is. 
Changes to the spec can be made and reviewed in separate PRs.

File: package.json
Comment (maurelian) [inline package.json:9]: @gakonst is this change sufficient to "add it to the global yarn workspace" as requested?

Comment (gakonst) [reply package.json:9]: Yes!

File: specs/package.json
Comment (gakonst) [inline specs/package.json:11]:
```suggestion
    "lint:fix": "yarn prettier --write \"{l2geth,protocol}/**/*.md\"",
    "lint": "yarn prettier --check \"{l2geth,protocol}/**/*.md\""
```

**Review by gakonst — APPROVED** [review]

File: specs/package.json
Comment (maurelian) [reply specs/package.json]: Nice, I'd considered exactly this change. :)

**Review by ben-chain — APPROVED** [review]
Woot woot, Let's Get This Merged!

**Pull request merged**
